pkg/skaffold/build/local/docker.go

```diff
@@ -46,7 +46,8 @@ func (b *Builder) buildDocker(ctx context.Context, out io.Writer, a *latest.Arti
 
 	var imageID string
 
-	if b.local.UseDockerCLI || b.local.UseBuildkit {
+	// TODO: Update DOC
```

Comment:
I think it might be easier to do something like the following:
  - if the user explicitly set `useBuildkit: true`, check if BuildKit is supported and if not then fail
  - if the user explicitly set `useBuildkit: false` then never use it
  - otherwise check if BuildKit is supported and use it if possible

So something like:
```
wantBuildKit := false
if b.local.UseBuildkit == nil || *b.local.UseBuildkit {
  if hasBuildkit := b.localDocker.HasBuildkitSupport(ctx); hasBuildkit {
    wantBuildkit = true
  } else if b.local.UseBuildkit != nil && *b.local.UseBuildkit {
     return "", errors.New("Docker does not support BuildKit")
  }
}
```

Reply:
if the user explicitly set `useBuildkit: true` and buildkit not supported, fail or fallback ?

Reply:
we should fail and not fallback.